lumicks/pylake/detail/image.py

```diff
@@ -192,40 +207,12 @@ def reconstruct_image(data, infowave, pixels_per_line, lines_per_frame=None, red
     np.ndarray
     """
     assert data.size == infowave.size
-
-    # Example infowave:
-    #  1 0 0 1 0 1 2 0 1 0 0 1 0 1 0 1 0 0 1 0 2 0 1 0 1 0 0 1 0 1 0 1 2 1 0 0 1
-    #              ^ <-----------------------> ^                       ^
-    #                       one pixel
-    valid_idx = infowave != InfowaveCode.discard
-    infowave = infowave[valid_idx]
-
-    # After discard:
-    #  1 1 1 2 1 1 1 1 1 2 1 1 1 1 1 2 1 1 1
-    #        ^ <-------> ^           ^
-    #         pixel_size (i.e. data samples per pixel)
-    # This should be:
-    #   pixel_sizes = np.diff(np.flatnonzero(infowave == InfowaveCode.pixel_boundary))
-    # But for now we assume that every pixel consists of the same number of samples
-    pixel_size = np.argmax(infowave) + 1
-
-    def round_up(size, n):
-        """Round up `size` to the nearest multiple of `n`"""
-        return int(math.ceil(size / n)) * n
-
-    resized_data = np.zeros(round_up(infowave.size, pixel_size))
-    resized_data[:infowave.size] = data[valid_idx]
+    subset, mask = process_infowave(infowave)
```

Comment:
Inevitable naming comment:
* `process_infowave()` is super generic. "Process" could be doing anything. I'd suggest `discard_zeros(infowave)`.
* I tend to get confused about which half is `mask` masking. IMO `valid_idx` was a bit clearer.